pkg/data/implementations/aws_remote_url.go

```diff
@@ -28,6 +28,7 @@ type s3Interface interface {
 // AWS-specific implementation of RemoteURLInterface
 type AWSRemoteURL struct {
 	s3Client        s3Interface
+	presignClient   s3Interface
```

Comment:
I want to deprecate this entire file and use the new stdlib's storage Api to get presigned URLs... Is it too much to do as part of this PR?

Reply:
ok, that makes sense. I think it's just a small change, will update this PR.